protocol/op-stack-manager-arch.md

```diff
@@ -0,0 +1,294 @@
+# Purpose
+
+OP Stack Manager is a contract that will deploy the L1 contracts for standard OP Stack chains in a single transaction.
+However, we also have use cases for deploying non-standard chains, such as deploying devnet and testnet contracts while developing interop.
+This document will outline an architecture that:
+
+- Enables the OP Stack Manager to deploy these non-standard chains as an initial milestone.
+- Gets us most of the way towards supporting standard chains deployments for the subsequent milestone.
+
+# Summary
+
+The original OP Stack Manager (OPSM) milestones were:
+
+- Milestone 1: Eliminating key handover complexity.
+- Milestone 2: Simplifying Superchain upgrades.
+- Milestone 3: Supporting interop upgrades.
+
+This design doc proposes the architecture for a Milestone 0.5 of "OPSM is used for all interop devnet & testnet L1 deployments".
+Adding this milestone will allow the interop team to build testing infrastructure that wraps OPSM, unifying our development, testnet, and production testing and deployment processes.
+
+# Problem Statement + Context
+
+*Some text copied from https://github.com/ethereum-optimism/design-docs/pull/52, which is related to this design doc.*
+
+The current L2 chain deployment approach originates from a time with Hardhat, single L1 target, and a single monolithic set of features.
+Since then the system has migrated to Foundry and extended for more features, but remains centered around a single monolithic deploy-config for all its features.
+
+The interop team needs a way to configure new multi-L2 deployments: The number of ways to compose L2s in tests grows past what a single legacy config template can support.
```

Comment:
I'm curious what some of these ways are. Are we talking about different chains interoperating with different features activated?

Reply:
The main issue is migrating into interop. Where some chains have their own different copy of some L1 contracts like the superchain config / implementations deployment. We want to be able to test how we get from those situations to the new system. And new tests / deployments following the op-stack-manager approach of deployment as much as possible.

And the secondary issue is that currently the L1 dev genesis creation itself is coupled to a single L2 deployment, so there's no composition of L2s in any form.